tools/tplg_parser/tplg_parser.c

```diff
@@ -929,7 +929,8 @@ int tplg_load_graph(int num_comps, int pipeline_id,
 		    int route_num, int count)
 {
 	struct snd_soc_tplg_dapm_graph_elem *graph_elem;
-	char *source, *sink;
+	char *source = "source not found";
+	char *sink   = "sink not found";
```

Comment:
I bet there will be analyzers/compilers that will complain that you assign `const char*` to `char*`, at least clang will for sure.

Reply:
So it'd be better to just init them to NULL, eventually you can later change printf IRT if they are NULL, but it's not worth the effort.

Reply:
I compiled this foo bar example with 3 different compilers and `-Wall -Wextra` . They all put the string in `.rodata` or `__cstrign` read-only section and all crash but none of them prints any warning! I guess there's just too much code out there in the world written like this :-)
```C
  char * foo = "bar";
  foo[1] = 'c';
```
Apple clang 10.0.1 
Fedora clang 9.0.1-2.fc31
gcc Red Hat 9.2.1-1

Another solution to move the string to a read-write section and prepare for a `-Wall -Wextra -WevenStrongerFuture` could be this:
```C
static char notfound[] = "not found";
char *source = notfound, *sink = notfound;
```
This avoids a conditional later near the `printf`
@ranj063 , @singalsu ?

Reply:
This is low value but time consuming. I really don't mind if we have conditional code in the printf, it's not hot code. At least the conditional would alert the user that there was an error in the topology.

Reply:
> This is low value but time consuming.

This -Werror=maybe-uninitialized PR is fixing the last `./scripts/host-build-all.sh`  build failure on my Fedora 31 system.

> I really don't mind if we have conditional code in the printf, it's not hot code. At least the conditional would alert the user that there was an error in the topology.

I suggested the `static char notfound[] = "not found";` (TESTED) because that's less code, so less time consuming :-) It would alert the user too, wouldn't it?

Reply:
yep that works for me too.

Reply:
@marc-hb @lgirdwood wow, sorry didnt realize this code was upstreamed!

Anyway, another alternative I'd suggest to avoid the error is 

```
if (!source || !sink || connection->source_id == -1 || connection->sink_id == -1) {
```

Reply:
or for that matter just this: 
```
if (!source || !sink)
```

Reply:
> if (!source || !sink)

This seemed like a winner so I just went ahead and did it, see last force push.

Reply:
> I guess there's just too much code out there in the world written like this :-)

Yea, I checked and that's the reason why it's allowed in C, to get error that I was talking about you'd have to compile with c++11 standard or above.